### PR TITLE
Brings DinoMod back online. Dinos are warm blooded, compies have been…

### PR DESCRIPTION
**Pull request by LyleSY**
Brings DinoMod back online. Dinos are warm blooded, compies have been nerfed to match turkeys. The dinos are in the sewers. Compies and eoraptors added to GROUP_SAFE monster group so they should pop up in some new places. The dinos must flow. -LyleSY

First pull request, fingers crossed.

### CODE REVIEW

File: data/mods/DinoMod/dinosaur.json [hunk 1]
```diff
@@ -870,11 +860,73 @@
       { "monster" : "mon_giant_crayfish", "freq" :  4, "cost_multiplier" : 0, "starts" : 720, "conditions" : ["DUSK", "NIGHT", "DAWN", "SPRING", "SUMMER", "AUTUMN"] },
       { "monster" : "mon_giant_crayfish", "freq" :  1, "cost_multiplier" : 0, "starts" : 792, "conditions" : ["SPRING", "SUMMER", "AUTUMN"] },
       { "monster" : "mon_giant_crayfish", "freq" :  4, "cost_multiplier" : 0, "starts" : 792, "conditions" : ["DUSK", "NIGHT", "DAWN", "SPRING", "SUMMER", "AUTUMN"] },
+      { "monster" : "mon_bear_smoky", "freq" :  1, "cost_multiplier" : 0, "starts" : 672, "conditions" : ["SPRING", "SUMMER", "AUTUMN"] },
```
Comment (BorkBorkGoesTheCode) [inline data/mods/DinoMod/dinosaur.json]: What does this have to do with Dinomod?

**Comment by DangerNoodle** [comment]
Why was this obsoleted anyway? I would guess the monster groups override? Unfortunately last I checked, `copy-from` will not allow for streamlining  monster groups.

**Comment by BorkBorkGoesTheCode** [comment]
Will any of the dinosaurs get feathers?

File: data/mods/DinoMod/dinosaur.json (hunk 1 above)
Comment (Zireael07) [inline data/mods/DinoMod/dinosaur.json]: Smoky bear AFAIK was an core monster but I'm not sure if it is still in core.

Comment (BrettDong) [inline data/mods/DinoMod/dinosaur.json]: It's still in `data/json/monsters.json` but is also in `data/json/default_blacklists.json`.

Comment (LyleSY) [inline data/mods/DinoMod/dinosaur.json]: Ha, fair enough. I checked the code in DinoMod and saw this change from vanilla, so refreshed from original since the intent of the mod isn't to hide smoky bears (which I have never seen in game).

**Comment by AlecWhite** [comment]
>Will any of the dinosaurs get feathers?

Dinosaurs as a source of feathers would allow for some Turok'esq killing with bow and arrows.

**Comment by BorkBorkGoesTheCode** [comment]
This mod really needs more dino spawn locations. I almost never see dinosaurs while playing, unless a field office appears. Maybe a dino-filled forest special could spawn in forests.

**Comment by Mecares** [comment]
I would like a Dino-Lab where the cloning went wrong

**Comment by LyleSY** [comment]
I like the way you people think. Yes, feathers are in the mod and it is awesome. It makes bringing down a big dinosaur pretty epic. Also, crazy amounts of meat and bone. I've been thinking about starting to creep them into the forest monster list also. Maybe just the herbivores and tiny predators? I like that the woods are relatively safe and don't want to mess that up.  Dino lab with a bunch of raptor mutagen and raptors has been much on my mind, though I've been worried about dinos fighting the zombies and robots. I'll look at the lab maps and see what can be done. Thanks!

**Comment by DangerNoodle** [comment]
I recall the days when the large predators make swamps unsafe, so I would reason mainly use the the less dangerous ones, yes.

**Comment by Zilenan91** [comment]
Dinos being able to spawn in swamps wouldn't really be much of a change for what we have right now. They're already warzones of dozens of creatures killing and eating eachother

**Comment by DangerNoodle** [comment]
I am referring to the fact that Dino Mod is set, or possibly used to be set(?) to add the T. Rex and Spinosaurus to swamps.

**Comment by Mecares** [comment]
Zombiefied TRex skeletons in Museums would be pretty awesome in my humble opinion.

**Comment by BorkBorkGoesTheCode** [comment]
>Zombiefied TRex skeletons in Museums would be pretty awesome in my humble opinion.

Don't you mean Z-Rex?

**Comment by LyleSY** [comment]
I dug into the src folder some to figure out how labs work. It looks like labs haven't been (can't be?) jsonized yet. Proper dino labs or dino lab rooms may have to wait until labs become more modder friendly. Dino basements or new specials like dino cabins seem more doable for now. As far as I can tell the only piece of labs in json is lab_notes.json

**Comment by kevingranade** [comment]
Hmm, it wouldn't really be that hard to make something like the lab (connected and tiled semirandom rooms) exposed to json, we should be able to come up with something.

For a relatively simple approach, the tiles (24x24 map areas) would have a static connection system, something like "north and west edges of tiles are walls except (0,12) and (13,0), which are doors" with a little extra logic to ensure the outer border of the entire system is sealed.  Then you can simply tile rooms according to some overall parameters like overall shape of the lab, size, depth, etc. More complex and interesting systems are certainly possible.

**Comment by DangerNoodle** [comment]
That would be useful for this, yes. Alternatively, a separate overmap for a different type of lab can be mocked up in JSON form, but it might need a minimum occurrences of zero.

**Comment by Zireael07** [comment]
@kevingranade: IIRC there was an issue about jsonizing labs/making them easier to mod, but can't find it atm...

**Comment by pisskop** [comment]
are compies dangerous?

group safe is just that.  be careful

**Comment by LyleSY** [comment]
@pisskop No, I nerfed their stats to match turkeys. Physically that seems to be the closest match.

**Comment by LyleSY** [comment]
Forest dino mod section is tested and ready. I'll wait for this pull to fly for the next bit.

File: data/mods/DinoMod/dinosaur.json (hunk 1 above)
Comment (Chezzo) [inline data/mods/DinoMod/dinosaur.json]: I think they are in Crazy Cataclysm mod.

**Comment by LyleSY** [comment]
Modified lab notes and snippets are tested and ready for the next version. I realized that similar to the monster lists, I have to include the core content or that stuff disappears when the mod is enabled.

**Comment by LyleSY** [comment]
Very cool, thanks all. I'll give folks some time to test before dropping the next version.
